components/pagination/collections/paging_collection.js

```diff
@@ -0,0 +1,369 @@
+ * one indexed interface to make consumption easier for views.
+ *
+ * Subclasses may want to override the following properties:
+ *      - url (string): The base url for the API endpoint.
+ *      - state (object): Object to overrride default state values
+ *        provided to Backbone.paginator.
+ *      - queryParams (object): Specifies Query parameters for the API
+ *        call using the Backbone.paginator API.  In the case of built-
+ *        in Backbone.paginator state keys, this maps those state keys
+ *        to query parameter keys.  queryParams can also map query
+ *        parameter keys to functions providing values for such keys.
+ *        Subclasses may add entries as necessary. By default,
+ *        'sort_order' is the query parameter used for sorting, with
+ *        values of 'asc' for increasing sort and 'desc' for decreasing
+ *        sort.
+ */
+;(function (define) {
+    'use strict';
+    define(['jquery', 'underscore', 'backbone.paginator'], function ($, _, PageableCollection) {
+        var PagingCollection = PageableCollection.extend({
+            mode: 'server',
+
+            isStale: false,
+
+            sortableFields: {},
+
+            searchString: '',
+            filterFields: {},
+
+            constructor: function (models, options) {
+                var defaultOptions = {
+                    state: {
+                        firstPage: 1,
+                        pageSize: 10,
+                        sortKey: null
+                    },
+                    queryParams: {
+                        currentPage: 'page',
+                        pageSize: 'page_size',
+                        sortKey: 'order_by',
+                        order: 'sort_order',
+                        text_search: function () { return this.searchString || null; }
+                    }
+                };
+
+                if (_.isUndefined(options)) { options = {}; }
+                _.extend(defaultOptions, _.omit(options, 'state', 'queryParams'));
+                _.extend(defaultOptions.state, options.state);
+                _.extend(defaultOptions.queryParams, options.queryParams);
+
+                PageableCollection.prototype.constructor.call(this, models, defaultOptions);
+            },
+
+            parse: function (response, options) {
+                // PageableCollection expects the response to be an
+                // array of two elements - the server-side state
+                // metadata (page, size, etc.), and the array of
+                // objects.
+                var modifiedResponse = [];
+                modifiedResponse.push(_.omit(response, 'results'));
+                modifiedResponse.push(response.results);
+                return PageableCollection.prototype.parse.call(this, modifiedResponse, options);
+            },
+
+            /* jshint unused:false */
+            parseState: function (response, queryParams, state, options) {
```

Comment:
Is this actually unused? If so, just remove it; otherwise could you leave a comment indicating why it's needed?